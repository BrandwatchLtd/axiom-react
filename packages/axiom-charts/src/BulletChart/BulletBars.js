import PropTypes from "prop-types";
import React, { Component } from "react";
import { Base } from "@brandwatch/axiom-components";
import Bar from "../Bar/Bar";
import Bars from "../Bar/Bars";
import { findBarGroupMax } from "./utils";

export default class BulletBars extends Component {
  static propTypes = {
    barLabel: PropTypes.number,
    direction: PropTypes.string,
    label: PropTypes.string,
    showBarLabel: PropTypes.bool,
    values: PropTypes.array.isRequired,
  };

  render() {
    const {
      direction,
      label,
      barLabel,
      showBarLabel,
      values,
      ...rest
    } = this.props;

    const barGroupMax = values.reduce(findBarGroupMax);

    return (
      <Base {...rest} className="ax-bullet-chart__block">
        <div className="ax-bullet-chart__bars">
          {values.map(({ color, value }) => (
            <div className="ax-bullet-chart__bars-bar" key={`${color}-div`}>
              <Bars direction={direction} label={label}>
                <Bar
                  color={color}
                  key={color}
                  label={`${barLabel}%`}
                  percent={value}
                  showLabel={value === barGroupMax && showBarLabel}
                />
              </Bars>
            </div>
          ))}
        </div>
      </Base>
    );
  }
}
