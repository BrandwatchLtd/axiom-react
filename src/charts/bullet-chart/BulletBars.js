import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Bar from '../bars/Bar';
import Bars from '../bars/Bars';
import Base from '../../components/base/Base';
import { findBarGroupMax, arrayOfLength } from './utils';

export default class BulletBars extends Component {
  static propTypes = {
    barLabel: PropTypes.number,
    direction: PropTypes.string,
    label: PropTypes.string,
    showBarLabel: PropTypes.bool,
    values: arrayOfLength.bind(null, 2),
  };

  render() {
    const {
      direction,
      label,
      barLabel,
      showBarLabel,
      values,
      ...rest } = this.props;

    const barGroupMax = values.reduce(findBarGroupMax);

    return (
      <Base { ...rest } className="ax-bullet-chart__block">
        <div className="ax-bullet-chart__bars">
          { values.map(({ color, value }) =>
            <div className="ax-bullet-chart__bars-bar" key={ `${color}-div` }>
              <Bars direction={ direction } label={ label }>
                <Bar
                    color={ color }
                    key={ color }
                    label={ `${barLabel}%` }
                    percent={ value }
                    showLabel={ value === barGroupMax && showBarLabel }/>
              </Bars>
            </div>
                )}
        </div>
      </Base>
    );
  }
}
