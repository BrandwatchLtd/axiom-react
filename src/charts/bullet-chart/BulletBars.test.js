import React from 'react';
import renderer from 'react-test-renderer';
import BulletBars from './BulletBars';
import { formattedData } from './BulletBars.test.fixture';

function getComponent(props = {}) {
  props.values = formattedData;

  return renderer.create(
    <BulletBars { ...props }>
    </BulletBars>
  );
}

describe('BulletBars', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with the correct label', () => {
    const props = {
      barLabel: 'label for highest value',
      showBarLabel: true,
    };
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
