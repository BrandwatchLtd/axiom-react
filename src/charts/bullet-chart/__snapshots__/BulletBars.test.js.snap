// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletBars renders with defaultProps 1`] = `
<div
  className="ax-bullet-chart__block"
>
  <div
    className="ax-bullet-chart__bars"
  >
    <div
      className="ax-bullet-chart__bars-bar"
    >
      <div
        className="ax-bars ax-bars--up"
      >
        <div
          className="ax-bars__bars-container"
        >
          <div
            className="ax-bars__bars"
          >
            <div
              className="ax-bars__bar"
              onClick={undefined}
              style={
                Object {
                  "height": "10%",
                  "width": false,
                }
              }
            >
              <div
                className="ax-bars__bar-rect ax-bars__bar-rect--giant-leap"
                style={
                  Object {
                    "height": false,
                    "minHeight": false,
                    "minWidth": "1rem",
                    "width": undefined,
                  }
                }
              />
              <div
                className="ax-bars__bar-label ax-bars__bar-label--hidden"
              >
                <small
                  className="ax-text--size-small"
                >
                  undefined%
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-bullet-chart__bars-bar"
    >
      <div
        className="ax-bars ax-bars--up"
      >
        <div
          className="ax-bars__bars-container"
        >
          <div
            className="ax-bars__bars"
          >
            <div
              className="ax-bars__bar"
              onClick={undefined}
              style={
                Object {
                  "height": "50%",
                  "width": false,
                }
              }
            >
              <div
                className="ax-bars__bar-rect ax-bars__bar-rect--critical-mass"
                style={
                  Object {
                    "height": false,
                    "minHeight": false,
                    "minWidth": "1rem",
                    "width": undefined,
                  }
                }
              />
              <div
                className="ax-bars__bar-label ax-bars__bar-label--hidden"
              >
                <small
                  className="ax-text--size-small"
                >
                  undefined%
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BulletBars renders with the correct label 1`] = `
<div
  className="ax-bullet-chart__block"
>
  <div
    className="ax-bullet-chart__bars"
  >
    <div
      className="ax-bullet-chart__bars-bar"
    >
      <div
        className="ax-bars ax-bars--up"
      >
        <div
          className="ax-bars__bars-container"
        >
          <div
            className="ax-bars__bars"
          >
            <div
              className="ax-bars__bar"
              onClick={undefined}
              style={
                Object {
                  "height": "10%",
                  "width": false,
                }
              }
            >
              <div
                className="ax-bars__bar-rect ax-bars__bar-rect--giant-leap"
                style={
                  Object {
                    "height": false,
                    "minHeight": false,
                    "minWidth": "1rem",
                    "width": undefined,
                  }
                }
              />
              <div
                className="ax-bars__bar-label ax-bars__bar-label--hidden"
              >
                <small
                  className="ax-text--size-small"
                >
                  1%
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-bullet-chart__bars-bar"
    >
      <div
        className="ax-bars ax-bars--up"
      >
        <div
          className="ax-bars__bars-container"
        >
          <div
            className="ax-bars__bars"
          >
            <div
              className="ax-bars__bar"
              onClick={undefined}
              style={
                Object {
                  "height": "50%",
                  "width": false,
                }
              }
            >
              <div
                className="ax-bars__bar-rect ax-bars__bar-rect--critical-mass"
                style={
                  Object {
                    "height": false,
                    "minHeight": false,
                    "minWidth": "1rem",
                    "width": undefined,
                  }
                }
              />
              <div
                className="ax-bars__bar-label"
              >
                <small
                  className="ax-text--size-small"
                >
                  1%
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
